### DISCUSSION TITLE
Web map title

### DISCUSSION THREAD

**Discussion by prushforth** [Maps4HTML/Web-Map-Custom-Element#558] [General]
When we listen to what a screen reader says about a map, I hear the page title and then "Application", "Interactive map", "Region", but I don't hear what the title of the map is.  This seems to me to be a shortcoming, I think each map should have a title (optionally) that is announced, and potentially visible somewhere, based on author  discretion.  

WDYT?

**Comment by Malvoz** [comment]
@opyh suggested something similar in https://github.com/Maps4HTML/HTML-Map-Element-UseCases-Requirements/issues/233.

However I think, much like with iframes, video, etc., this should already be possible through authors providing an `aria-label` or `title`.

Though there may be problems with _if_ or _when_ an author-provided `title` or `aria-label` is announced however, because (again) [it is the leaflet container that is focused and not the map element](https://github.com/Maps4HTML/experiments/pull/66#pullrequestreview-789867588) (IIRC this is why we have `role` on both the leaflet container and the map element).

**Reply by Malvoz** [reply]
And if the author wants a visible title for the map they could do something like:

```html
<figure>
  <figcaption>Map of something</figcaption>
  <mapml-viewer>
    ...
  </mapml-viewer>
</figure>

// or
<section>
  <hX>Map of something</hX>
  <mapml-viewer>
    ...
  </mapml-viewer>
</section>
```

**Comment by prushforth** [comment]
@ben-lu-uw it's a discussion at present, we can convert to an issue if we need to fix some stuff.

**Comment by Malvoz** [comment]
Related to this discussion, @MapBox has been experimenting(?) with Dynamic Text Descriptions: https://youtu.be/GYWf4Zz4tsE?t=372 👀

**Reply by Malvoz** [reply]
Which may help answer [Ed's question](https://www.w3.org/community/maps4html/2022/09/28/why-spatial-is-very-special/#:~:text=Ed%20Parsons%20raised,of%20such%20labelling.):

> Ed Parsons raised what I consider to be a profound question about the Denver map example: What is it about the map, besides being visually obvious to someone familiar with United States geography, that tells us that it’s a map of Denver?  Even the text list representation of the map might not indicate that these are features from Denver, and the map itself even might not be labelled as such, despite the importance of such labelling.

**Reply by prushforth** [reply]
Yep.  Using reverse geocoding seems to me to be the most valuable techique there.  Even a purely image-based map could leverage such a technique.  Maybe this could be part of a user preference?  I.e. you put the url of a reverse geocoding service into your browser settings, adjust a few settings and boom, you've got a dynamic map reading experience.  This relates to the issues we encountered with this [experiment](https://maps4html.org/experiments/screenreader/announce-zoom/), reads meaningless (coordinates-related) info.  What's required is something meaningful to an individual user, which might be where the preference comes in. 

Further, an idea that is lurking is that in a map-and-location enabled Web, [search itself](https://www.google.ca/search?q=foo) could be effectively "reverse geocoding" i.e. supply a location, receive relevance-ranked Web pages as features for any given purpose.

**Comment by prushforth** [comment]
Update:  @kevinkim31 implemented a [\<map-caption> custom element](https://maps4html.org/web-map-doc/docs/elements/map-caption/) that allows an author to describe the purpose / location of the map.  

> . This relates to the issues we encountered with this [experiment](https://maps4html.org/experiments/screenreader/announce-zoom/), reads meaningless (coordinates-related) info. What's required is something meaningful to an individual user, which might be where the preference comes in.

We removed the reading of coordinates, as I considered that a "failed" experiment.  We kept the announcing of zoom level, as that may be somewhat useful, once a user becomes used to zoom levels.  

Kevin added a customised version of L.Control.Scale which announces the scale ratio [as best as we can calculate it](http://discourse.wicg.io/t/one-inch-is-not-an-inch/1228/40?u=peterr) on moveend.
